source/particle/output/interface.cc

```diff
@@ -49,18 +51,71 @@ namespace aspect
       template <int dim>
       template <class Archive>
       void Interface<dim>::serialize (Archive &ar, const unsigned int)
-      {}
+      {
+        ar &output_file_suffix
+        &file_index
+        ;
+      }
 
       template <int dim>
       void
-      Interface<dim>::save (std::ostringstream &) const
-      {}
+      Interface<dim>::save (std::map<std::string, std::string> &status_strings) const
+      {
+        std::ostringstream os;
+        aspect::oarchive oa (os);
+
+        oa << (*this);
+
+        status_strings["Particles output"] = os.str();
+      }
 
       template <int dim>
       void
-      Interface<dim>::load (std::istringstream &)
-      {}
+      Interface<dim>::load (std::map<std::string, std::string> &status_strings)
+      {
+        // see if something was saved
+        if (status_strings.find("Particles output") != status_strings.end())
+          {
+            std::istringstream is (status_strings.find("Particles output")->second);
+            aspect::iarchive ia (is);
+
+            ia >> (*this);
+          }
+      }
+
+      template <int dim>
+      std::string
+      Interface<dim>::get_file_name () const
+      {
+        return "particles-"
+               + get_file_index()
+               + "."
+               + Utilities::int_to_string(Utilities::MPI::this_mpi_process(this->get_mpi_communicator()), 4)
+               + "."
+               + output_file_suffix;
+      }
+
+      template <int dim>
+      std::string
+      Interface<dim>::get_particle_output_location () const
+      {
+        return this->get_output_directory()
+               + "particles/";
+      }
 
+      template <int dim>
+      std::string
+      Interface<dim>::get_file_index () const
+      {
+        return Utilities::int_to_string(file_index, 5);
+      }
+
+      template<int dim>
+      void
+      Interface<dim>::increment_file_index ()
+      {
+        file_index++;
```

Comment:
use `++file_index;`. Not important in this case, but we try to be consistent about this.